### PR TITLE
Add CUDA graphs CI to torchbench CI job

### PR DESCRIPTION
**Pull request by ezyang**
Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

**Review by jansel — CHANGES_REQUESTED** [review]
I think you need to update the pytorch version pin:
https://github.com/pytorch/torchdynamo/blob/4f5a24e566a0b1d231aaca110f4ee75eaf8dc935/Makefile#L12

**Comment by ezyang** [comment]
this is no longer mergeable as is, need https://github.com/pytorch/torchdynamo/pull/704 first. I hacked out the broken bits for now

**Comment by ezyang** [comment]
this is not mergeable as is, but if people have opinions about the color of the bikeshed (e.g., run cuda graphs ci alongside coverage; do it as a parallel job; etc) please let me know

File: torchdynamo/optimizations/python_key.py [hunk 1]
```diff
@@ -140,16 +136,17 @@ def run_node(self, n: torch.fx.Node):
...
-            python_tensor_cls(elem, proxy)
-            for elem, proxy in zip(chain(params_flat, example_inputs), proxy_args)
-        ]
-        with _stateless.reparametrize_module(
-            gm, pytree.tree_unflatten(args[:params_len], params_spec)
-        ):
-            out = PatchingInterpreter(gm).run(*args[params_len:])
-
-        assert isinstance(out, (tuple, list)), "graph must output tuple()"
+        with proxy_mode:
```
Comment (msaroufim) [inline torchdynamo/optimizations/python_key.py:139]: Maybe it's late but this context manager is defined as `None` above so shouldn't have an `__enter__` defined?

File: Makefile [hunk 2]
```diff
@@ -110,7 +107,7 @@ build-deps: clone-deps
 	(cd ../torchtext   && python setup.py clean && python setup.py develop)
 	(cd ../torchaudio  && python setup.py clean && python setup.py develop)
 	(cd ../detectron2  && python setup.py clean && python setup.py develop)
-	(cd ../functorch   && python setup.py clean && python setup.py develop)
+	(cd ../pytorch/functorch   && python setup.py clean && python setup.py develop)
```
Comment (msaroufim) [inline Makefile:110]: So people still need to install functorch even though it's back in core?

File: torchdynamo/optimizations/python_key.py [hunk 3]
```diff
@@ -160,6 +157,7 @@ def unpack(x):
 
     with pythonkey_decompose({**aot_autograd_decompositions, **decompositions}):
         tracer: torch.fx.Tracer = PythonKeyTracer()
+        proxy_mode = ProxyTorchDispatchMode(tracer)
```
Comment (msaroufim) [inline torchdynamo/optimizations/python_key.py:160]: does this change the state of the tracer?

File: Makefile (hunk 2 above)
Comment (ezyang) [reply Makefile:110]: we don't have consolidated build yet, so yes you still have to explicitly build it

File: torchdynamo/optimizations/python_key.py (hunk 1 above)
Comment (ezyang) [reply torchdynamo/optimizations/python_key.py:139]: these are from the version update PR, go put your comments in that one

**Pull request closed without merging**